src/app/view/editor-tools-view-component/editor-tools-view.component.ts

```diff
@@ -420,41 +439,79 @@ export class EditorToolsViewComponent {
     };
   }
 
-  private getOriginDestinationContainerToExport(): ContainertoExportData {
-    const htmlElementToExport = document.getElementById("main-origin-destination-container");
-    if (htmlElementToExport === null) {
-      return undefined;
+  private exportOriginDestinationCanvasToPNG(filename: string) {
+    const sel = d3.select("#main-origin-destination-canvas");
+    const canvas = sel.node(); // <- real HTMLCanvasElement or null
+
+    if (!canvas) {
+      console.error("Canvas nicht gefunden:", sel);
+    } else {
+      const ctx = canvas.getContext("2d");
+      console.log("Canvas found, size:", canvas.width, canvas.height, ctx);
     }
-    const bbox = (htmlElementToExport as unknown as SVGGElement).getBBox();
-    const padding = 10;
-    const param = {
-      encoderOptions: 1.0,
-      scale: 1.0,
-      left: bbox.x - padding,
-      top: bbox.y - padding,
-      width: bbox.width + 2 * padding,
-      height: bbox.height + 2 * padding,
-      backgroundColor: this.uiInteractionService.getActiveTheme().backgroundColor,
-    };
 
-    const essentialProps = [
-      "fill",
-      "stroke",
-      "stroke-width",
-      "stroke-dasharray",
-      "font-family",
-      "font-size",
-      "font-weight",
-      "opacity",
-      "text-anchor",
-      "dominant-baseline",
-    ];
+    if (canvas === null) {
+      return;
+    }
 
-    return {
-      documentToExport: htmlElementToExport,
-      exportParameter: param,
-      essentialProps: essentialProps,
+    // quality only used for image/jpeg
+    const dataUrl = canvas.toDataURL("image/png");
+    const a = document.createElement("a");
+    a.href = dataUrl;
+    a.download = filename;
+    document.body.appendChild(a);
+    a.click();
+    a.remove();
```

Comment:
```suggestion
    a.remove();
    setTimeout(() => URL.revokeObjectURL(url), 2000);
```

you can/should revoke here